sdf.md

```diff
@@ -1816,10 +1818,72 @@ The initial set of registrations is described in {{sdftype-r}}.
 Security Considerations {#seccons}
 =======================
 
-Some wider issues are discussed in {{-seccons}}.
-
-(Specifics: TBD.)
-
+Some wider security considerations applicable to Things are discussed
+in {{-seccons}}.
+{{Section 5 of -cddl}} gives an overview over security considerations
+that arise when formal description techniques are used to govern
+interoperability; analogs of these security considerations can apply
+to SDF.
+
+The security considerations of underlying building blocks such as
+those detailed in {{Section 10 of -utf8}} apply.
+SDF uses JSON as a representation language; {{-json}} for a number of
+cases indicates that implementation behavior for certain constructs
+allowed by the JSON grammar is unpredictable.
+Implementations need to be robust against invalid or unpredictable
+cases on input, preferably by rejecting input that is invalid or
+that would lead to unpredictable behavior, and need to avoid generating
+these cases on output.
+
+Implementations of model languages may also exhibit
+performance-related availability issues when the attacker can control
+the input, see {{Section 4.1 of -jsonpath}} for a brief discussion.
+
+SDF may be used in two processes that are often security relevant:
+*validation* of data that is intended to be described by SDF models, and
+*augmentation* of these data with information obtained from the SDF
+models that apply.
+
+Implementations need to ascertain the provenance and applicability of
+the SDF models they employ operationally in these ways.
+Implementations that make use of the composition mechanisms defined in this
+document need to do this for each of the components they combine
+into the SDF models they employ.
+Essentially, this process needs to undergo the same care and scrutiny
+as any other introduction of source code into a build environment; the
+possibility of supply-chain attacks on the modules imported needs to
+be considered.
+
+Specifically, implementations that rely on model-based input
+validation for enforcing certain properties of the data structure
+ingested (which, if not validated, could lead to malfunctions such as
+crashes and remote code execution) need to be particularly careful
+about the data models they apply, including their provenance and
+potential changes of these properties that upgrades to the referenced
+modules may (inadvertently or as part of an attack) cause.
+More generally speaking, implementations should strive to be robust
+against expected and unexpected limitations of the model-based input
+validation mechanisms and their implementations.
+
+Similarly, implementations that rely on model-based augmentation may
```

Comment:
Perhaps useful to add some words to explain what we mean why "model-based augmentation"?

Reply:
The fourth paragraph was intended to do that.  I added "model-based" twice.